lib/kcl.js

```diff
@@ -153,41 +154,70 @@ module.exports = function(config, kinesis) {
       return delete instanceShardList[shard.id];
     }
 
-    // expose the checkpoint function to the procerssRecords, so they can checkpoint
+    // expose the checkpoint function to the processRecords, so they can checkpoint
     // without requesting more records.
     shard.checkpointFunc = db.checkpoint.bind(null, shard);
 
-    kinesis.getRecords(
-      { ShardIterator: shard.iterator, Limit: config.limit },
-      function(err, resp) {
-        if (err && err.code === 'SyntaxError') return setTimeout(getRecords.bind(this, shard), 100);
-        if (err) throw err;
+    if (config.verbose) logger.info('[ getRecords ] params:', JSON.stringify({
+      ShardIterator: shard.iterator,
+      Limit: config.limit
+    }));
 
-        if (config.cloudwatchNamespace &&
-          resp.Records &&
-          resp.Records[0] &&
-          resp.Records[0].ApproximateArrivalTimestamp) {
-          throttledPutToCloudwatch('ShardIteratorAgeInMs', (+new Date()) - resp.Records[0].ApproximateArrivalTimestamp, 'Milliseconds', shard.id);
-        }
+    var req = kinesis.getRecords({ShardIterator: shard.iterator, Limit: config.limit});
+    var requestTimeOut = setTimeout(req.abort.bind(req), 5000);
 
-        // malformed responses will have an undefined NextShardIterator. In that case, we keep the existing one.
-        shard.iterator = resp.NextShardIterator === undefined ? shard.iterator : resp.NextShardIterator;
-        shard.lastGetRecords = +new Date();
+    req.on('error', function (err) {
+      if (err.code === 'SyntaxError' || err.code === 'RequestAbortedError') {
+        if (config.verbose) logger.info('[ getRecords ] Found SyntaxError', err);
+        return setTimeout(getRecords.bind(this, shard), 500);
+      }
+      if (err.code === 'ProvisionedThroughputExceededException') {
+        if (config.verbose) logger.info('[ getRecords ] Found ProvisionedThroughputExceededException', err);
+        return setTimeout(getRecords.bind(this, shard), 1000);
```

Comment:
This code retries by calling `getRecords()` function again.  That isnt the same the AWS SDK retry.  That will start a new request object and sent again.  So do both retry? Should we be setting retries to 0 on this request since we are handling them ourselves?